src/bsp/platform/stm32fx_nucleo_wifi/xi_bsp_io_net_stm32fx_nucleo_wifi.c

```diff
@@ -179,37 +193,30 @@ xi_bsp_io_net_state_t xi_bsp_io_net_select( xi_bsp_socket_events_t* socket_event
 
     size_t socket_id = 0;
 
-    if ( xi_net_state.wifi_state == xi_wifi_state_connected )
+    for ( socket_id = 0; socket_id < socket_events_array_size; ++socket_id )
     {
-        for ( socket_id = 0; socket_id < socket_events_array_size; ++socket_id )
-        {
-            xi_bsp_socket_events_t* socket_events = &socket_events_array[socket_id];
-
-            if ( 1 == socket_events->in_socket_want_read )
-            {
-                if ( NULL != xi_net_state.head )
-                {
-                    socket_events->out_socket_can_read = 1;
-                }
-            }
+        xi_bsp_socket_events_t* socket_events = &socket_events_array[socket_id];
 
-            if ( 1 == socket_events->in_socket_want_connect )
+        if ( 1 == socket_events->in_socket_want_read )
+        {
+            if ( NULL != xi_net_state.head )
             {
-                socket_events->out_socket_connect_finished = 1;
+                socket_events->out_socket_can_read = 1;
             }
+        }
 
-            if ( 1 == socket_events->in_socket_want_write )
-            {
-                socket_events->out_socket_can_write = 1;
-            }
+        if ( 1 == socket_events->in_socket_want_connect )
+        {
+            socket_events->out_socket_connect_finished = 1;
         }
 
-        return XI_BSP_IO_NET_STATE_OK;
-    }
-    else
-    {
-        return XI_BSP_IO_NET_STATE_ERROR;
+        if ( 1 == socket_events->in_socket_want_write )
+        {
+            socket_events->out_socket_can_write = 1;
+        }
     }
+
+    return XI_BSP_IO_NET_STATE_OK;
```

Comment:
Yes that's true. I was thinking about it - but at the same time we don't want to block the user application for longer that it is necessary - this not having select situation is really problematic and we have to carefully think about it - because of few facts:
Fact no 1. - if we wait for too long - the WiFi cheap may read too much data and it may say - no more memory closing 
Fact no 2. - if we wait for too long - the WiFi cheap may read big amount of data - and before Xively will consume it it will fill our heap with chunks 
Fact no 2. - if we wait for too short - we will consume more power.... in MQTT protocol there is no way to save power or go to sleep mode between sending and receiving messages but the MQTT-SN is ready for it. 

So with that being said I think we should aim into something between ( 50ms - 250ms ) question what will work the best if users want to implement their own activities - I think that it will be much more clear once we add the sensor board and CPM connection so we will be able to see what is the response time of the demo application.